moveit_experimental/moveit_jog_arm/src/collision_check_thread.cpp

```diff
@@ -113,15 +118,20 @@ void CollisionCheckThread::startMainLoop(JogArmShared& shared_variables)
 
     // If we are far from a collision, velocity_scale should be 1.
     // If we are very close to a collision, velocity_scale should be ~zero.
-    // When collision_proximity_threshold is breached, start decelerating exponentially.
-    else if (collision_distance < parameters_.collision_proximity_threshold)
+    // When scene_collision_proximity_threshold is breached, start decelerating exponentially.
+    else if (scene_collision_distance < parameters_.scene_collision_proximity_threshold)
     {
       // velocity_scale = e ^ k * (collision_distance - threshold)
       // k = - ln(0.001) / collision_proximity_threshold
       // velocity_scale should equal one when collision_distance is at collision_proximity_threshold.
       // velocity_scale should equal 0.001 when collision_distance is at zero.
-      velocity_scale =
-          exp(velocity_scale_coefficient * (collision_distance - parameters_.collision_proximity_threshold));
+      velocity_scale = exp(scene_velocity_scale_coefficient *
+                           (scene_collision_distance - parameters_.scene_collision_proximity_threshold));
+    }
+    else if (self_collision_distance < parameters_.self_collision_proximity_threshold)
```

Comment:
I think I wouldn't split both distance checks as if/else. Instead I would control for the lower velocity scale, even if this probably doesn't have that many practical implications.

Reply:
I did think through this and thought I'd avoided any bad outcomes, but on second thought it's not very maintainable this way. Instead I just made three discrete if blocks, using `velocity_scale = std::min(velocity_scale, ...);` in each, so we always use the slowest result.